File: agent-plugin/plugin/src/main/java/org/matilda/template/TemplatePluginComponent.java
package org.matilda.template;

import dagger.Component;
import org.matilda.commands.CommandRegistry;
import org.matilda.commands.PluginDependenciesModule;
import org.matilda.template.generated.commands.CommandRegistryModule;

import javax.inject.Singleton;

@Component(modules = {CommandRegistryModule.class, PluginDependenciesModule.class})
@Singleton
public interface TemplatePluginComponent {
    CommandRegistry commandRegistry();
}

File: agent-plugin/plugin/src/commonMain/kotlin/org/matilda/template/TemplatePluginComponent.kt
package org.matilda.template

import me.tatarka.inject.annotations.Component
import org.matilda.commands.CommandRegistry
import org.matilda.commands.MatildaScope
import org.matilda.commands.PluginDependenciesComponent
import org.matilda.template.generated.commands.CommandRegistryModule

@Component
@MatildaScope
abstract class TemplatePluginComponent(@Component val pluginDependencies: PluginDependenciesComponent) : CommandRegistryModule {
    abstract fun commandRegistry(): CommandRegistry
}
